packages/core/test/new-api/stored-deployment-serializer.ts

```diff
@@ -632,21 +744,31 @@ describe("stored deployment serializer", () => {
 function assertSerializableModuleIn(deployment: StoredDeployment) {
   const serialized = JSON.stringify(
     StoredDeploymentSerializer.serialize(deployment),
+    // This is not actually needed, but we use it to be able to compare the
+    // serialized string, which can be easier to debug.
     sortedKeysJsonStringifyReplacer,
     2
   );
 
+  const deserialized = StoredDeploymentDeserializer.deserialize(
+    JSON.parse(serialized)
+  );
+
   const reserialized = JSON.stringify(
-    StoredDeploymentSerializer.serialize(
-      StoredDeploymentDeserializer.deserialize(JSON.parse(serialized))
-    ),
+    StoredDeploymentSerializer.serialize(deserialized),
     sortedKeysJsonStringifyReplacer,
     2
   );
 
   assert.equal(
     serialized,
     reserialized,
+    "Module serialization not the same across serialization/deserialization"
+  );
+
+  assert.deepEqual(
```

Comment:
I really wanted to add this test, as it was hard to convince myself that this worked correctly.

This lead me to changing the serialization to preserve the order of a few things.